Core/include/Acts/Geometry/GeometryIdentifier.hpp

```diff
@@ -124,6 +130,9 @@ class GeometryIdentifier {
 
 std::ostream& operator<<(std::ostream& os, GeometryIdentifier id);
 
+using GeometryIdentifierHook =
+    std::function<GeometryIdentifier(GeometryIdentifier, const Surface&)>;
```

Comment:
I like it :+1: apart from changing the identifier it should also be easy to build a mapping between acts and experiment IDs

Reply:
> So, if we had lets say 20 surfaces front size and 20 surfaces backside, would you do 0-20 and a side bit, but 0-40 and a side bit ?

Answered in person: second one will keep the current behaviour ... Good.